e2etests/node_scan_rhcos_test.go

```diff
@@ -52,65 +50,77 @@ var vulnTar = &v1.Vulnerability{
 	Severity: "Moderate",
 }
 
+func buildRequestCase(notes []v1.Note) *v1.GetNodeVulnerabilitiesRequest {
+	// From: https://www.redhat.com/security/data/metrics/repository-to-cpe.json
+	// "rhel-8-for-x86_64-appstream-rpms": {"cpes": ["cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream"]},
+	// "rhel-8-for-x86_64-baseos-rpms": {"cpes": ["cpe:/o:redhat:enterprise_linux:8::baseos", "cpe:/o:redhat:rhel:8.3::baseos"]}
+	cpes := []string{
+		"cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream",
+		"cpe:/a:redhat:enterprise_linux:8::baseos", "cpe:/a:redhat:rhel:8.3::baseos",
+	}
+	return &v1.GetNodeVulnerabilitiesRequest{
+		OsImage:          "Red Hat Enterprise Linux CoreOS 45.82.202008101249-0 (Ootpa)",
+		KernelVersion:    "0.0.1",
+		KubeletVersion:   "0.0.1",
+		KubeproxyVersion: "0.0.1",
+		Runtime: &v1.GetNodeVulnerabilitiesRequest_ContainerRuntime{
+			Name:    "docker",
+			Version: "0.0.1",
+		},
+		Notes: notes,
+		Components: &v1.Components{
+			Namespace: "rhcos:4.11",
+			RhelComponents: []*v1.RHELComponent{
+				{
+					Id:        int64(1),
+					Name:      "libksba",
+					Namespace: "rhel:8",
+					Version:   "1.3.5-7.el8",
+					Arch:      "x86_64",
+					Module:    "", // must be empty, otherwise scanner does not return any vulns
+					Cpes:      cpes,
+					AddedBy:   "",
+				},
+				{
+					Id:        int64(2),
+					Name:      "tar",
+					Namespace: "rhel:8",
+					Version:   "1.27.1.el8",
+					Arch:      "x86_64",
+					Module:    "",
+					Cpes:      cpes,
+					AddedBy:   "",
+				},
+				{
+					Id:        int64(3),
+					Name:      "grep",
+					Namespace: "rhel:8",
+					Version:   "3.1-6.el8",
+					Arch:      "x86_64",
+					Module:    "",
+					Cpes:      cpes,
+					AddedBy:   "",
+				},
+			},
+			LanguageComponents: nil,
+		},
+	}
+}
+
 func TestGRPCGetRHCOSNodeVulnerabilities(t *testing.T) {
 	conn := connectToScanner(t)
 	client := v1.NewNodeScanServiceClient(conn)
-	if !features.RHCOSNodeScanning.Enabled() {
-		t.Skipf("Feature RHCOS Node Scanning is disabled")
-	}
 
 	cases := []struct {
+		name             string
```

Comment:
Please switch to `map[string]struct`.

Reply:
This tiny improvement can still be done.

Reply:
Here: [d26da56](https://github.com/stackrox/scanner/pull/1004/commits/d26da561cb06ab166a9ffb23919e904972ad8b93)